pir/pir-internal/src/main/java/com/duckduckgo/pir/internal/common/BrokerStepsParser.kt

```diff
@@ -76,8 +101,19 @@ class RealBrokerStepsParser @Inject constructor(
         stepsJson: String,
     ): BrokerStep? = withContext(dispatcherProvider.io()) {
         return@withContext runCatching {
-            adapter.fromJson(stepsJson)?.apply {
-                this.brokerName = brokerName
+            adapter.fromJson(stepsJson)?.run {
+                if (this is OptOutStep) {
+                    this.copy(
+                        brokerName = brokerName,
+                        profilesToOptOut = repository.getExtractProfileResultForBroker(brokerName)?.extractResults?.filter {
```

Comment:
This step is only relevant for optout. We need to obtain the extracted profiles with result set to true (if any) - those are the considered valid records.